trieTools/zeroBalanceSystemAccountChecker/extraTokensCrossChecker.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"errors"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	maxRequestsRetrial = 10
+	multipleSearchBulk = 10000
+)
+
+type tokenBalancesGetter interface {
+	getBalance(address, token string) (string, error)
+}
+
+type elasticMultiSearchClient interface {
+	GetMultiple(index string, requests []string) ([]byte, error)
+}
+
+type extraTokensChecker struct {
+	nftBalancesGetter tokenBalancesGetter
+	elasticClient     elasticMultiSearchClient
+}
+
+func newExtraTokensCrossChecker(client elasticMultiSearchClient, nftBalancesGetter tokenBalancesGetter) (crossTokenChecker, error) {
+	if client == nil {
+		return nil, errors.New("nil elastic client provided")
+	}
+	if nftBalancesGetter == nil {
+		return nil, errors.New("nil nft balances getter provided")
+	}
+
+	return &extraTokensChecker{
+		nftBalancesGetter: nftBalancesGetter,
+		elasticClient:     client,
+	}, nil
+}
+
+func (etc *extraTokensChecker) crossCheckExtraTokens(tokens map[string]struct{}) ([]string, error) {
+	numTokens := len(tokens)
+	log.Info("starting to cross-check", "num of tokens", numTokens)
+
+	bulkSize := core.MinInt(multipleSearchBulk, numTokens)
+	tokensThatStillExist := make([]string, 0)
+	requests := make([]string, 0, bulkSize)
+	currTokenIdx := 0
```

Comment:
please rename this variable (no shortcuts) + the next one. What does "ctRequests" should mean?

Reply:
- `curr` = current
- `ct` = counter

These are one of the most used shortcuts in coding, especially `ct`.
Renamed:
- `currTokenIdx` -> `tokenIdx`
- `ctReqyests` -> `numRequests`

Reply:
because we generally do not use shortcuts so the code reads better.

Reply:
Renamed them